corehq/apps/reports/util.py

```diff
@@ -524,20 +527,25 @@ def update_tableau_user(domain, username, role=None, groups=[]):
     Update the TableauUser object to have the given role and new group details. The `groups` arg should be a list
     of TableauGroupTuples.
     '''
-    session = TableauAPISession.create_session_for_domain(domain)
-    user = TableauUser.objects.filter(
-        server=session.tableau_connected_app.server
-    ).get(username=username)
-    if role:
-        user.role = role
-    user.save()
-    _update_user_remote(session, user, groups)
+    try:
```

Comment:
try to make the try block as isolated to the possible even if that means doing 2 try/except blocks